### ISSUE TITLE
ValueError/TypeError with path dependency

### ISSUE THREAD

**Issue by apnewberry** [sdispater/poetry#454]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Ubuntu 18.04
- **Poetry version**: 0.11.5
- **toml** below

## Issue

**Note:** the dependency uses setup.py and Pipfile/Pipfile.lock.

```toml
[tool.poetry]

name="my_demo"
version = "0.0.1"
description = ""
authors = ["me"]

[tool.poetry.dependencies]
python = "^3.6"

epype = {path = "/home/user/dotfiles/dotfiles/.emacs.d/python/epype"}

```

```python
$ poetry lock -vvv
Using virtualenv: /home/user/.cache/pypoetry/virtualenvs/epype-py3.6
Updating dependencies
Resolving dependencies...
   1: fact: my-demo is 0.0.1
   1: derived: my-demo
   1: fact: my-demo depends on epype (0.0.1)
   1: selecting my-demo (0.0.1)
   1: derived: epype (0.0.1)
   1: selecting epype (0.0.1)
   1: Version solving took 0.004 seconds.
   1: Tried 1 solutions.

                                   
[ValueError]        
Invalid type <class 'NoneType'>  
                                   
Exception trace:
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/cleo/application.py in run() at line 94
   status_code = self.do_run(input_, output_)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/poetry/console/application.py in do_run() at line 88
   return super(Application, self).do_run(i, o)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/cleo/application.py in do_run() at line 197
   status_code = command.run(input_, output_)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/poetry/console/commands/command.py in run() at line 77
   return super(BaseCommand, self).run(i, o)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/cleo/commands/base_command.py in run() at line 146
   status_code = self.execute(input_, output_)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/cleo/commands/command.py in execute() at line 107
   return self.handle()
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/poetry/console/commands/lock.py in handle() at line 33
   return installer.run()
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/poetry/installation/installer.py in run() at line 76
   self._do_install(local_repo)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/poetry/installation/installer.py in _do_install() at line 277
   self._package, local_repo.packages
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/poetry/packages/locker.py in set_lock_data() at line 122
   lock["package"] = packages
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/tomlkit/container.py in __setitem__() at line 488
   self.append(key, value)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/tomlkit/container.py in append() at line 90
   item = _item(item)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/tomlkit/items.py in item() at line 55
   i = item(_v)
 /home/user/.envs/dotfiles-_bp8OBTv/lib/python3.6/site-packages/tomlkit/items.py in item() at line 77
   raise ValueError("Invalid type {}".format(type(value)))

lock

```

**Comment by songww** [comment]
Hello
Looks like exception accurred when resolve dependencies of  `/home/user/dotfiles/dotfiles/.emacs.d/python/epype`.

**Comment by apnewberry** [comment]
I switched `epype` to use poetry instead of setup.py and that worked, but I wish the error could be more informative. I still don't know what the problem was.

**Comment by bryanforbes** [comment]
I hit this issue with a [git package](https://github.com/mkurnikov/pytest-mypy-plugins/blob/master/setup.py) I was using. I traced it down to the package not having a `description` in its `setup.py`. It ended up as `None` coming out of [`pkginfo.UnpackedSDist` in `puzzle/provider.py`](https://github.com/sdispater/poetry/blob/94304e9daac7b20d755877e94a05197c19b81670/poetry/puzzle/provider.py#L338) and then when the package info is passed to a [`TomlDocument` in `packages/locker.py`](https://github.com/sdispater/poetry/blob/94304e9daa/poetry/packages/locker.py#L135) it raises an error because `TomlDocument` doesn't handle `None`.

@sdispater If you could give me a hint as to how to fix this, I could take a stab at it. I'm not sure how exactly `None` should be handled in these cases.

**Comment by sdispater** [comment]
@bryanforbes I think that this is something that should be handled by TOMLKit. `None` (and its TOML equivalent `null`) is valid and should be properly handled.

**Comment by sdispater** [comment]
I thought there was support for `null` in TOML, my bad.

So, I think it should be handled in the locker: leave the `description` line out if it's null and handle the absence of `description` when loading data from the lock file.

**Comment by sdispater** [comment]
This should be fixed in the latest `0.12.11` release.
